perfectsense/brightspot-cms
# BSP-2500: RTC refactor

Changes RTC to use `AsyncContext` from the servlet spec (http://docs.oracle.com/javaee/6/api/javax/servlet/AsyncContext.html) directly instead of the Atmosphere library, which had subtle issues that were difficult to troubleshoot due to how Brightspot had to use it.

Some things to try out to test in the new implementation:

Normal use case:

* Open up same content with 2 different users
* Change a field in user A and make sure that the field locking works on user B

Disconnect:

* Open up same content with 2 different users
* Make sure that the viewers show up in the publishing widget for both users
* Go offline in user A (e.g. in Chrome, this can be done via Developer Tools > Network > Offline)
* See that the viewer disappears from user B within about 15 seconds

Connection sharing

* Open up the dashboard with 2 different users
* Open up a content in user A by right clicking and opening in new window/tab
* Open up the same content in user B by right clicking and opening in a new window/tab
* Make sure that the viewers should up in the dashboard and in the publishing widget for both users
* Close the dashboard in user A
* Make sure that the viewer from the publishing widget is still visible in user B
